apis/bigquerydatatransfer/v1alpha1/bigquerydatatransferconfig_types.go

```diff
@@ -0,0 +1,197 @@
+)
+
+var BigQueryDataTransferConfigGVK = GroupVersion.WithKind("BigQueryDataTransferConfig")
+
+// BigQueryDataTransferConfigSpec defines the desired state of BigQueryDataTransferConfig
+// +kcc:proto=google.cloud.bigquery.datatransfer.v1.TransferConfig
+type BigQueryDataTransferConfigSpec struct {
+	// The number of days to look back to automatically refresh the data.
+	//  For example, if `data_refresh_window_days = 10`, then every day
+	//  BigQuery reingests data for [today-10, today-1], rather than ingesting data
+	//  for just [today-1].
+	//  Only valid if the data source supports the feature. Set the value to 0
+	//  to use the default value.
+	DataRefreshWindowDays *int32 `json:"dataRefreshWindowDays,omitempty"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="DataSourceID field is immutable"
+	// Immutable.
+	// Data source ID. This cannot be changed once data transfer is created. The
+	//  full list of available data source IDs can be returned through an API call:
+	//  https://cloud.google.com/bigquery-transfer/docs/reference/datatransfer/rest/v1/projects.locations.dataSources/list
+	DataSourceID *string `json:"dataSourceID,omitempty"`
+
+	// The BigQuery target dataset id.
+	DestinationDatasetID *string `json:"destinationDatasetID,omitempty"`
+
+	// Is this config disabled. When set to true, no runs will be scheduled for
+	//  this transfer config.
+	Disabled *bool `json:"disabled,omitempty"`
+
+	// User specified display name for the data transfer.
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Email notifications will be sent according to these preferences
+	//  to the email address of the user who owns this transfer config.
+	EmailPreferences *EmailPreferences `json:"emailPreferences,omitempty"`
+
+	// The encryption configuration part. Currently, it is only used for the
+	//  optional KMS key name. The BigQuery service account of your project must be
+	//  granted permissions to use the key. Read methods will return the key name
+	//  applied in effect. Write methods will apply the key if it is present, or
+	//  otherwise try to apply project default keys if it is absent.
+	EncryptionConfiguration *EncryptionConfiguration `json:"encryptionConfiguration,omitempty"`
+
+	// Pub/Sub topic where notifications will be sent after transfer runs
+	//  associated with this transfer config finish.
+	//
+	//  The format for specifying a pubsub topic is:
+	//  `projects/{project_id}/topics/{topic_id}`
+	NotificationPubsubTopic *string `json:"notificationPubsubTopic,omitempty"`
+
+	// Parameters specific to each data source. For more information see the
+	//  bq tab in the 'Setting up a data transfer' section for each data source.
+	//  For example the parameters for Cloud Storage transfers are listed here:
+	//  https://cloud.google.com/bigquery-transfer/docs/cloud-storage-transfer#bq
+	Params map[string]string `json:"params,omitempty"`
+
+	// +required
+	Parent `json:",inline"`
```

Comment:
Thank you for using this style!  It looks nicer in code than in the guide 😆

Reply:
I found it in your doc! Maybe we should add `Parent` to the controller builder template?